score/score.py

```diff
@@ -0,0 +1,255 @@
+    if save.ascension_level > 0:
+        # Only applies to the score from the following bonuses:
+        # Floors Climbed
+        # Enemies Killed
+        # Elites Killed
+        # Bosses Slain
+        # Champion
+        # Perfect
+        # Beyond Perfect
+        # Overkill
+        # C-c-c-combo.
+        bonuses_total = sum(bonus.score_bonus for bonus in [
+            get_floors_climbed_bonus(save),
+            get_enemies_killed_bonus(save),
+            get_act1_elites_killed_bonus(save),
+            get_act2_elites_killed_bonus(save),
+            get_act3_elites_killed_bonus(save),
+            get_bosses_slain_bonus(save),
+            get_champions_bonus(save),
+            get_overkill_bonus(save),
+            get_combo_bonus(save)
+        ])
+        asc_bonus.score_bonus = round(bonuses_total * 0.05 * save.ascension_level)
+    return asc_bonus
+
+def get_floors_climbed_bonus(save: Savefile) -> Score:
+    """Get 5 points for each floor climbed."""
+    bonus = Score("Floors Climbed", save.current_floor - 1, should_show=True)
+    bonus.score_bonus = 5 * (save.current_floor - 1) # will need to do more testing if this is current-1 or just current
+    return bonus
+
+def get_enemies_killed_bonus(save: Savefile) -> Score:
+    """Get 2 points for each monster defeated."""
+    bonus = Score("Enemies Killed", save.monsters_killed, should_show=True)
+    bonus.score_bonus = 2 * save.monsters_killed
+    return bonus
+
+def get_act1_elites_killed_bonus(save: Savefile) -> Score:
+    """Get 10 points for each Act 1 elite killed."""
+    bonus = Score("Exordium Elites Killed", save.act1_elites_killed, should_show=True)
+    bonus.score_bonus = 10 * save.act1_elites_killed
+    return bonus
+
+def get_act2_elites_killed_bonus(save: Savefile) -> Score:
+    """Get 20 points for each Act 2 elite killed."""
+    bonus = Score("City Elites Killed", save.act2_elites_killed, should_show=save.act_num >= 2)
+    bonus.score_bonus = 20 * save.act2_elites_killed
+    return bonus
+
+def get_act3_elites_killed_bonus(save: Savefile) -> Score:
+    """Get 30 points for each Act 3 elite killed."""
+    bonus = Score("Beyond Elites Killed", save.act3_elites_killed, should_show=save.act_num >= 3)
+    bonus.score_bonus = 30 * save.act3_elites_killed
+    return bonus
+
+def get_champions_bonus(save: Savefile) -> Score:
+    """Get 25 points for each elite perfected."""
+    bonus = Score("Champion", save.perfect_elites)
+    bonus.score_bonus = 25 * save.perfect_elites
+    return bonus
+
+def get_bosses_slain_bonus(save: Savefile) -> Score:
+    """Get 25 points for each boss killed."""
+    boss_nodes = sum(1 for node in save.path if node.room_type == "Boss")
+    bonus = Score("Bosses Slain", boss_nodes, should_show=True)
+    score = 0
+    match boss_nodes:
+        case 1:
+            score = 50
+        case 2:
+            score = 150
+        case 3:
+            score = 300
+        case 4:
+            score = 500
+        case 5:
```

Comment:
This should be `case _:` so that it catches everything that is 5 or more (so that glitches and double-node don't break it). This should then have a `case 0:` at the top.